lib/cequel/metal/keyspace.rb

```diff
@@ -311,17 +321,17 @@ def cluster
       private :lock
 
       def client_retry
-        retries = max_retries
+        retries_remaining = max_retries
         begin
           yield
         rescue Cassandra::Errors::NoHostsAvailable,
                Cassandra::Errors::ExecutionError,
-               Cassandra::Errors::TimeoutError
-          clear_active_connections!
-          raise if retries == 0
-          retries -= 1
-          sleep(retry_delay)
-          retry
+               Cassandra::Errors::TimeoutError => exc
+          # Pass the error to the error policy. It can reraise,
+          # if it returns at all it indicates a retry should occur               
+          error_policy.handle_error(self, exc, retries_remaining)
```

Comment:
It seems like retrying should be a feature of the error policy. Maybe sometimes one wouldn't want to retry. What about moving all of this functionality into the error policy as a `#execute_stmt(&blk)` method that would look very similar to the current `#client_retry`